src/components/Explore/TokenDetails/TokenDetail.tsx

```diff
@@ -176,15 +178,34 @@ const TruncatedAddress = styled.span`
     display: flex;
   }
 `
+const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: string }>`
+  display: flex;
+  border-radius: 5px;
+  padding: 4px 6px;
+  align-items: center;
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 12px;
+  color: ${({ theme, networkColor }) => networkColor ?? theme.textPrimary};
+  background-color: ${({ theme, backgroundColor }) => backgroundColor ?? theme.backgroundSurface};
+`
 
 export default function LoadedTokenDetail({ address }: { address: string }) {
   const theme = useTheme()
+  const NETWORK_BADGE_COLORS: Record<string, string> = {
```

Comment:
this is slightly ugly, lmk if you have ideas on how to do this better (not sure how to 'access' theme outside CSS/component)